android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/WorkerBinder.kt

```diff
@@ -178,11 +272,54 @@ private fun <T : Comparable<T>> Worker.bind(
    * GlobalScope won't leak the job, because the flow completes when lifecycle completes.
    */
   @OptIn(DelicateCoroutinesApi::class)
-  val job = GlobalScope.launch(RibDispatchers.Unconfined, start = CoroutineStart.UNDISPATCHED) {
+  val job = GlobalScope.launch(
+    coroutineScopeLauncher.coroutineDispatcher,
+    start = coroutineScopeLauncher.coroutineStart
+  ) {
     lifecycle
       .takeWhile { it < lifecycleRange.endInclusive }
-      .onCompletion { onStop() }
-      .collect { onStart(WorkerScopeProvider(lifecycle.asScopeProvider(lifecycleRange))) }
+      .onCompletion {
+        val totalBindingDurationMilli = measureTimeMillis { onStop() }
+
+        workerBinderDurationListeners.reportDuration(
+          this@bind,
+          WorkerEvent.STOP,
+          totalBindingDurationMilli
+        )
+      }
+      .collect {
+        val workerScopeProvider = WorkerScopeProvider(lifecycle.asScopeProvider(lifecycleRange))
+        val totalBindingDurationMilli = measureTimeMillis { onStart(workerScopeProvider) }
+
+        workerBinderDurationListeners.reportDuration(
+          this@bind,
+          WorkerEvent.START,
+          totalBindingDurationMilli
+        )
+      }
```

Comment:
For simplicity we could make an extension like:
```kotlin
private inline fun Worker.reportDuration(event: WorkerEvent, block: Worker.() -> Unit) {
  val duration = measureTimeMillis { block() }
  workerBinderDurationListeners.reportDuration(
      this@bind,
      WorkerEvent.STOP,
      totalBindingDurationMilli
   )
}
```
And we can use like:
```kotlin
    lifecycle
      .takeWhile { it < lifecycleRange.endInclusive }
      .onCompletion { reportDuration(WorkerEvent.STOP) { onStop() } }
      .collect { 
        reportDuration(WorkerEvent.START) { 
          onStart(WorkerScopeProvider(lifecycle.asScopeProvider(lifecycleRange)))
        }
      }
```

Reply:
thanks! this looks so much cleaner. Will apply soon